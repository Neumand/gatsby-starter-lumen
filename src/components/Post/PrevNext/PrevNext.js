import classNames from 'classnames/bind';
import { Link } from 'gatsby';
import React from 'react';
import styles from './PrevNext.module.scss';

type Props = {
  next: {
    title: string,
    slug: string,
  },
  prev: {
    title: string,
    slug: string,
  },
};

const cx = classNames.bind(styles);

const PrevNext = ({ next, prev }: Props) => {
  const prevClassName = cx({
    'prev-next__prev-link': true,
    'prev-next__prev-link--disable': !prev,
  });

  const nextClassName = cx({
    'prev-next__next-link': true,
    'prev-next__next-link--disable': !next,
  });

  return (
    <div className={styles['prev-next']}>
      <div className={styles['prev-next__prev']}>
        <Link to={prev ? prev.slug : '/'} className={prevClassName}>
          {prev ? `← ${prev.title}` : ' '}
        </Link>
      </div>
      <div className={styles['prev-next__next']}>
        <Link to={next ? next.slug : '/'} className={nextClassName}>
          {next ? `${next.title} →` : ' '}
        </Link>
      </div>
    </div>
  );
};

export default PrevNext;
