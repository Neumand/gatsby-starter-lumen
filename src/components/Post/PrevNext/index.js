// @flow strict
export { default } from './PrevNext';
