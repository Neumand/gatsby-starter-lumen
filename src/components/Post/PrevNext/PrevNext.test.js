import React from 'react';
import renderer from 'react-test-renderer';
import PrevNext from './PrevNext';

describe('PrevNext', () => {
  const props = {
    next: {
      title: 'Test Post 3',
      slug: '/posts/test-post-3',
    },
    prev: {
      title: 'Test Post 1',
      slug: '/posts/test-post-1',
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PrevNext {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
