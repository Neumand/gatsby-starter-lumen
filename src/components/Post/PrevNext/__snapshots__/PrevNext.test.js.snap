// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevNext renders correctly 1`] = `
<div
  className="prev-next"
>
  <div
    className="prev-next__prev"
  >
    <a
      className="prev-next__prev-link"
      href="/posts/test-post-1"
    >
      ← Test Post 1
    </a>
  </div>
  <div
    className="prev-next__next"
  >
    <a
      className="prev-next__next-link"
      href="/posts/test-post-3"
    >
      Test Post 3 →
    </a>
  </div>
</div>
`;
